docs/readmes/proposals/p006_subscriber_state_view.md

```diff
@@ -0,0 +1,166 @@
+For federated deployments where the subscriber configurations are federated to HSS or the policy configurations are federated to PCRF, the Orc8r does not have any visibility into the configurations received.
+
+
+## General Goal
+
+The goal here is to export consolidated session state owned by SessionD to the Orc8r so that we can 
+
+1. View all existing session states, uniquely identified by IMSI+APN 
+2. View runtime state of active policies + QoS
+
+
+
+## Proposed State from SessionD → GW State service
+
+Here is the state Protobuf type:
+
+```
+message State {
+    // Type determines how the value is deserialized and validated on the cloud service side
+    string type = 1; // 'subscriber_state'
+    string deviceID = 2; // IMSI'
+    // Value contains the operational state json-serialized.
+    bytes value = 3; // JSON serialized Subscriber State type below
+    uint64 version = 4; // ?
+}
+```
+
+Here is the blob value:
+
+```
+  subscriber_state:
+    type: object
+    description: An object that maps IMSI->APN->SubscriberState
+    properties:
+      imsi:
+        type: string
+      sessions:
+        type: array
+        items:
+          $ref: '#/definitions/policy_rule'
+  session_state:
+    type: object
+    description: Used to describe a session's state per APN
+    additionalProperties:
+      type: object
+      properties:
+        apn:
+          type: string
+        msisdn:
+          type: string
+        session_id:
+          type: string
+        ipv4:
+          type: string
+        session_start_time:
+          type: uint32
+        lifecycle_state: 
+          type: string
+        active_duration_sec:
+          type: uint32
+        active_policy_rules:
+          type: array
+          items:
+            $ref:# A JSON-serialized version of protobuf PolicyRule
+```
+
+### Notes on the subscriber_state fields
+
+* IMSI+APN uniquely identifies each active session
+    * In SessionD there could be situations where we multiple sessions for the 
+    same IMSI+APN at the same time, but there is only one ACTIVE session at a time. 
+    Additionally, we do not keep around non-ACTIVE sessions for longer than 5 seconds typically. 
+* MSISDN value is propagated from MME to SessionD on session creation. The value is fetched from HSS/SubscriberDB. 
+* This is the first round of fields suggested, but the following fields could be useful for debugging as well
+    * Attach time & session duration in seconds (Update: Included now)
+
+**[P1]** As with other gateway states reported by the state service, SessionD’s state will also be sent as an untyped JSON blob. These subscriber states will be reported from the Access network. (LTE/FEG_LTE/CWF etc.)
+
+## Proposed Orc8r API Endpoints
+
+**[P1]** Define read-only `lte` subscriber state endpoints that dumps the most recently reported subscriber state. 
```

Comment:
We don't need to define new endpoints -- subscriber state is already read-only (it's not part of the `MutableSubscriber` model). Instead we can
- use the existing `/lte/{network_id}/subscribers/` endpoint
- append to the subscriber state types [0] the new state types we want to expose

That change should be sufficient for the MVP of this doc

[0] lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go:89